lib/spack/spack/cmd/install.py

```diff
@@ -167,6 +167,8 @@ def add_cdash_args(subparser, add_help):
 Defaults to spec of the package to install."""
         cdash_help['site'] = """The site name that will be reported to CDash.
 Defaults to current system hostname."""
+        cdash_help['authtoken'] = """Path to file containing an authentication
+token to present to CDash"""
```

Comment:
Given that secrets such as these tokens are usually passed through env vars, and not through files, wouldn't it make more sense to just have `--cdash-auth-token-var` and pass the env var name?  Or, better yet, just predefine a name like `SPACK_CDASH_AUTH_TOKEN` and have users set that secret up in their CI.  I think that would be smoother given the way most CI systems I'm familiar with work.  Thoughts?

Reply:
That would indeed remove some code from the CI code, as I currently have to read it from the environment, then write it to a temporary file and pass it on the cli.

Reply:
Thanks for the suggestion. I updated this PR to read the auth token from an environment variable instead. PTAL when you get a chance & let me know if you think any further changes are needed.